### DISCUSSION TITLE
CronTrigger Miss When Running On Multiple Instances Simultaneously

### DISCUSSION THREAD

**Discussion by jaderowl** [quartznet/quartznet#2548] [Q&A] [answered]
Hi there,
I’m experiencing an issue with CronTrigger when running multiple instances of Quartz.NET. It seems like when two instances of my application are running in parallel, some triggers get missed without logging any errors. I have load balancing enabled, and both instances are connected to the same SQL Server database for job storage.
```
services.AddQuartz(q =>
{
    q.UseMicrosoftDependencyInjectionJobFactory();

    // Adding the job
    q.AddJob<SampleJob>(opts => opts.WithIdentity("SampleJob"));

    // Adding the trigger
    q.AddTrigger(opts => opts
        .ForJob("SampleJob")
        .WithCronSchedule("0/30 * * * * ?") // Every 30 seconds
        .WithMisfireHandlingInstructionFireAndProceed());
});

services.AddQuartzServer(options => options.WaitForJobsToComplete = true);
```

**Expected Behavior**: The job should fire exactly every 30 seconds without misfires across multiple instances

**Actual Behavior**: sometimes the job does not execute, and no error is logged

Environment:

Quartz.NET version: 3.4.0
Database: SQL Server 2019
Hosting environment: ASP.NET Core 7.0
Running in Docker containers

Has anyone else experienced something similar? Any advice on how to prevent this behavior?

**Comment by lahma** [comment]
You configuration doesn't contain any database or clustering configuration.

**Reply by jaderowl** [reply]
how do i apply it then

**Reply by raghadmta** [reply, marked as answer]
set  _**UseSqlServer**_ and _**UseClustering**_ option in Quartz.
```
 q.UseSqlServer(options =>
{
    options.ConnectionString = Configuration.GetConnectionString("QuartzDatabase");
    options.UseClustering();
});
```

**Reply by jaderowl** [reply]
Thanks everyone it works
